### PR TITLE
Update minimum Dask requirement to 2021.6.0

### LINKED ISSUE #8457: [BUG] Can't import dask_cudf in latest cudf nightlies
```
(rapids) rgelhausen@rl-dgx-r11-u36-rapids-dgx101:~$ conda list | grep rapids
# packages in environment at /home/rgelhausen/conda/envs/rapids:
cudf                      21.08.00a210607 cuda_11.2_py38_gae8ee8a060_118    rapidsai-nightly
dask-cuda                 21.08.00a210608         py38_20    rapidsai-nightly
dask-cudf                 21.08.00a210607 py38_gae8ee8a060_118    rapidsai-nightly
libcudf                   21.08.00a210608 cuda11.2_gaa82646248_120    rapidsai-nightly
librmm                    21.08.00a210608 cuda11.2_gaa2a2f3_18    rapidsai-nightly
rmm                       21.08.00a210608 cuda_11.2_py38_gaa2a2f3_18    rapidsai-nightly
ucx                       1.9.0+gcd9efd3       cuda11.2_0    rapidsai-nightly
ucx-proc                  1.0.0                       gpu    rapidsai-nightly
ucx-py                    0.21.0a210608   py38_gcd9efd3_14    rapidsai-nightly

conda list | grep dask
dask                      2021.5.1           pyhd8ed1ab_0    conda-forge
dask-core                 2021.5.1           pyhd8ed1ab_0    conda-forge
dask-cuda                 21.08.00a210608         py38_20    rapidsai-nightly
dask-cudf                 21.08.00a210607 py38_gae8ee8a060_118    rapidsai-nightly
```

```
iPython 3.8.10 | packaged by conda-forge | (default, May 11 2021, 07:01:05) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dask_cudf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rgelhausen/conda/envs/rapids/lib/python3.8/site-packages/dask_cudf/__init__.py", line 6, in <module>
    from . import backends
  File "/home/rgelhausen/conda/envs/rapids/lib/python3.8/site-packages/dask_cudf/backends.py", line 38, in <module>
    from .core import DataFrame, Index, Series
  File "/home/rgelhausen/conda/envs/rapids/lib/python3.8/site-packages/dask_cudf/core.py", line 17, in <module>
    from dask.dataframe.core import (
ImportError: cannot import name 'make_meta' from 'dask.dataframe.core' (/home/rgelhausen/conda/envs/rapids/lib/python3.8/site-packages/dask/dataframe/core.py)
```

### PR DESCRIPTION
**Pull request by pentschev**
We need Dask >= 2021.6.0 to resolve the `make_meta` issue fixed in https://github.com/dask/dask/pull/7743 , thus updating minimum Dask/Distributed requirement.

Fixes #8457

### CODE REVIEW

**Comment by pentschev** [comment]
cc @randerzander @quasiben @jakirkham

**Comment by quasiben** [comment]
In this PR are you trying to change the dev environments, the releases, nightlies, or all three ?  

Note that for CI testing we do not use any pins:
https://github.com/rapidsai/cudf/pull/8425

**Comment by pentschev** [comment]
All three, that's because we need to ensure package versions, whereas CI is installing from source.

**Comment by quasiben** [comment]
The last two releases we've been pinning a maximum version of dask/distributed.  When we get to the 21.08 we will probably need to change things to restrict to a new maximum

**Comment by pentschev** [comment]
Right, but 21.08 is still unreleased, do we want to pin to a max version _during_ development?

**Comment by jakirkham** [comment]
cc @galipremsagar

**Comment by jakirkham** [comment]
We could also just pin to this Dask + Distributed release for now and slowly relax that as new releases come out

**Comment by quasiben** [comment]
I think it's ok to unpin during development.  But I think it would be good to hear from @galipremsagar  and/or @shwina

**Comment by shwina** [comment]
Should be fine to unpin during development.

**Review by quasiben — APPROVED** [review]

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
We'll need `integration` repo changes too: https://github.com/rapidsai/integration, but would probably need to hold off until `branch-21.08` is created in that repo.

**Review by jakirkham — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

**Comment by quasiben** [comment]
If we merged this PR while we waited for the integration repo issue to be resolved what would happen ?  Nightly Docker Images would still be broken ?  Anything else ?

**Comment by pentschev** [comment]
> If we merged this PR while we waited for the integration repo issue to be resolved what would happen ? Nightly Docker Images would still be broken ? Anything else ?

My understanding is this will unblock at least those environments that don't necessarily depend on other packages that may encounter the same type of issue this PR is trying to solve. I'm not sure what would happen to the Docker images, but I think this shouldn't introduce any new issues.

With the above said, should we merge this? We have several approvals, but it's not entirely clear to me if we should hold until integration creates branch-21.08.

**Comment by ajschmidt8** [comment]
`branch-21.08` has been created in the `integration` repo
https://github.com/rapidsai/integration/tree/branch-21.08

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**
